lib/src/entrypoint.dart

```diff
@@ -642,4 +646,127 @@ class Entrypoint {
     ensureDir(p.dirname(newPath));
     renameDir(oldPath, newPath);
   }
+
+  /// Execute any outstanding Dart scripts in `after_install`.
+  Future runAfterInstallScripts(SolveResult result) async {
+    // Count how many scripts we could potentially run.
+    var allScripts = <String, List<String>>{};
+
+    for (var package in result.changedPackages) {
+      var scripts = result.pubspecs[package].afterInstall;
+      if (scripts.isNotEmpty) allScripts[package] = scripts;
+    }
+
+    // If no scripts need to be run, don't bother updating the cache.
+    if (allScripts.isEmpty) return;
+
+    // Figure out what the last time every script was run.
+    // If a script has not changed since the last time it was run,
+    // Don't run it.
+    var systemScriptCache = await AfterInstallCache.load(cache.rootDir);
+    var localScriptCache = p.equals(cache.rootDir, cachePath)
+        ? systemScriptCache
+        : await AfterInstallCache.load(cachePath);
+    var mergedCache = systemScriptCache.merge(localScriptCache);
+
+    // Sort so that the root's scripts run last.
+    var changedPackages = result.changedPackages.toList();
+    changedPackages.sort((a, b) => a == root.name ? 1 : a.compareTo(b));
+
+    // Run scripts in dependencies.
+    for (var package in changedPackages) {
+      var pkg = package == root.name
+          ? root
+          : cache.load(result.packages
+              .firstWhere((a) => a.name == package, orElse: () => null));
+
+      for (var script in pkg.pubspec.afterInstall) {
+        var absolutePath = p.normalize(p.absolute(p.join(pkg.dir, script)));
+        var scriptFile = new File(absolutePath);
+
+        // Don't run the script against there are new changes.
+        if (!await mergedCache.isOutdated(absolutePath)) continue;
+
+        if (!await scriptFile.exists()) {
+          if (pkg == root)
+            log.warning(log.yellow(
+                '"after_install" script "$absolutePath" does not exist.'));
+        } else {
+          // Never trust any third-party scripts. Show a prompt before running scripts.
+          //
+          // The user can opt to print the contents of the script.
+          YesNoPrintResponse response = YesNoPrintResponse.yes;
+
+          if (pkg != root) {
+            var message =
+                '\npackage:$package wants to run the script "$absolutePath".';
+
+            do {
+              response = await confirmOrPrint(message);
```

Comment:
We would need to figure out how to handle this on CI systems like travis - or within editors like VsCode/Intellij

Reply:
That's a valid concern, and something I didn't consider at all. At least for Travis, in theory a command-line arg or env variable could be used to auto-deny (or maybe auto-allow) every script?

Reply:
Yes - I think the default behavior would need to be to auto-deny, with a flag or environment variable to auto-allow.

Reply:
should we consider being able to whitelist scripts from certain packages?

Reply:
> should we consider being able to whitelist scripts from certain packages?

I like that idea